lmdeploy/pytorch/engine/logits_process.py

```diff
@@ -93,6 +96,30 @@ def _multinomial_sampling(scores: torch.Tensor,
     return multinomial_sampling(scores, seeds, offsets, indices)
 
 
+def _guided_sampling(response_formats: Tuple[Dict], scores: torch.Tensor,
+                     guided_input_ids: torch.Tensor, tokenizer: object):
```

Comment:
Skip sampling if guided_input_ids is None to avoid unnecessary loop.

Reply:
guided_input_ids is a tuple of dict or None, since some requests might require guided decoding while others do not.

Reply:
guided_input_ids is a Tensor or None.